### PR TITLE
[PHP 8.0] Pre-attributes - update annotations to named arguments

### PR DESCRIPTION
**Pull request by TomasVotruba**
Just making use of `@NamedArgumentConstructor` annotation: https://github.com/doctrine/annotations/blob/1.11.x/docs/en/custom.rst#optional-constructors-with-named-parameters

This will make https://github.com/apitte/core/pull/161 much easier to add :+1:  **Merge this PR befor that one**

<br>

I found it by miracle today in here :) 
https://github.com/koriym/Koriym.Attributes#compatible-annotation

### CODE REVIEW

File: src/Annotation/Controller/Id.php
Comment (TomasVotruba) [inline src/Annotation/Controller/Id.php:25]: There checks are now replaced by native PHP and required arguments without default value in constructor :+1:

**Comment by TomasVotruba** [comment]
Ready for review and merge :+1:

File: src/Annotation/Controller/RequestParameter.php
Comment (f3l1x) [inline src/Annotation/Controller/RequestParameter.php:46]: `in` should be before required. It's more important.

File: tests/cases/Annotation/Controller/RequestParameters.phpt
Comment (f3l1x) [inline tests/cases/Annotation/Controller/RequestParameters.phpt:37]:
```suggestion
			new RequestParameters([
```

**Review by f3l1x — CHANGES_REQUESTED** [review]

**Comment by f3l1x** [comment]
PIng @TomasVotruba

File: src/Annotation/Controller/RequestParameter.php
Comment (TomasVotruba) [reply src/Annotation/Controller/RequestParameter.php]: Done :+1:

File: composer.json
Comment (TomasVotruba) [inline composer.json:29]: - Since 1.12 the `Doctrine\Common\Annotations\Annotation\NamedArgumentConstructor` class is present, but that's not enough.
- Since 1.13 the sole agument counts as default (with no array) = it finally behaves like attributes since this version



For more, I wrote a post about this process → https://tomasvotruba.com/blog/doctrine-annotations-and-attributes-living-together-in-peace/

**Comment by TomasVotruba** [comment]
Oh crap, the CI needs your approval again :facepalm: Could you click it?

I hope this will disappear once you merge at least one of my PRs and I'll become "verified" real user :laughing:

**Comment by TomasVotruba** [comment]
The least I could do is to run PHP 7.2 single failed test `DI/ApiExtension.phpt` locally. It's fixed now :+1:

**Comment by f3l1x** [comment]
Merged, thanks.

**Pull request merged**
